query/src/main/java/org/infinispan/query/logging/Log.java

```diff
@@ -204,5 +204,11 @@ public interface Log extends BasicLogger {
    @Message(value = "Reindexing starting.", id = 14063)
    void indexingStarting();
 
+   @Message(value = "Multiple knn predicates are not supported at the moment.", id = 14064)
+   ParsingException multipleKnnPredicates();
```

Comment:
Is this because of a bug we should be aware of?

I mean, multiple knn predicates probably doesn't make much *sense*, but I'd have expected it to just work.

Reply:
The problem is related to the broadcast queries. If we allow multiple knn predicates or in general having a boolean predicate having a knn as a leaf predicate, finding the expected number is not so easy. For now we're going to support only a single knn predicate with the possibility of using the internal clause to filter the elements on which to apply the knn search. We eventually evaluate to support this option, only if other data systems support it.